### ISSUE TITLE
[BUG] crc.exe process spawned by Daemon Scheduled Task is not killed within crc cleanup on Windows 11

### ISSUE THREAD

**Issue by adrianriobo** [crc-org/crc#3715]
When trying to run crc on a `Windows 11 Pro` edition it ends up failing due to wrong state for the crc daemon. 

Steps:

1. run `crc setup` (50% of times it will end up failing when checking the state for the crc daemon)
2. if `crc setup` finish ok run `crc start` it fails b/c crc daemon is not running

Checking the task scheduler we can see the task is `Ready`. It says it will be triggered on log on ... (so in case we reboot the VM) then the task appears as Running.

Also this exact flow has been tested on a `Windows 11 Enterprise` edition and on `Windows 10 Pro` and it is working so far, after running the `crc setup` we can see the crc daemon Running (as expected).

As a possible solution we can try to add a 2nd trigger on the task spec, the 2nd trigger would be type: `At task creation/modification`

**Comment by adrianriobo** [comment]
The previous issue is directly related with having a crc.exe process for the daemon running. Under that condition the Task status stay as Ready.

One possible reason for the situation is having crc running previously and the trying to cleanup everything (`crc cleanup`).

On windows 11 when running `crc cleanup` the crc.exe for the daemon is not killed.

**Comment by anjannath** [comment]
o/p from when `crc cleanup` hangs on windows 11:
```
PS C:\Users\rhqp> crc cleanup --log-level debug
DEBU CRC version: 2.20.0+f3a947
DEBU OpenShift version: 4.13.0
DEBU Podman version: 4.4.4
DEBU Running 'crc cleanup'
INFO Removing the daemon task
DEBU Running 'Get-ScheduledTask -TaskName crcDaemon'
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: Get-ScheduledTask : No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'crcDaemon'.  Verify the
value of the property and retry.
At line:1 char:43
+ ... reference = 'SilentlyContinue'; Get-ScheduledTask -TaskName crcDaemon
+                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (crcDaemon:String) [Get-ScheduledTask], CimJobException
    + FullyQualifiedErrorId : CmdletizationQuery_NotFound_TaskName,Get-ScheduledTask

DEBU crcDaemon task is not installed: exit status 1 : Get-ScheduledTask : No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'crcDaemon'.  Verify the
value of the property and retry.
At line:1 char:43
+ ... reference = 'SilentlyContinue'; Get-ScheduledTask -TaskName crcDaemon
+                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (crcDaemon:String) [Get-ScheduledTask], CimJobException
    + FullyQualifiedErrorId : CmdletizationQuery_NotFound_TaskName,Get-ScheduledTask

INFO Removing CRC Specific entries from user's known_hosts file
INFO Removing hosts file records added by CRC
INFO Removing pull secret from the keyring
INFO Removing older logs
INFO Removing CRC Machine Instance directory
DEBU Deleting machines directory
DEBU Failed to delete crc machines directory: remove C:\Users\rhqp\.crc\machines\crc\crc\Virtual Machines\9CD71052-C950-4AFB-9F51-3B6B5840F004.vmcx: The process cannot access the file because it is being used by another process.
INFO Removing crc's virtual machine
DEBU Running 'Get-VM -Name crc'
DEBU Running 'Stop-VM -Name "crc" -Force'
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: Stop-VM : 'crc' failed to change state.
The operation cannot be performed while the object is in its current state.
At line:1 char:43
+ $ProgressPreference = 'SilentlyContinue'; Stop-VM -Name "crc" -Force
+                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Stop-VM], VirtualizationException
    + FullyQualifiedErrorId : InvalidState,Microsoft.HyperV.PowerShell.Commands.StopVM

DEBU exit status 1
Failed to delete crc machines directory: remove C:\Users\rhqp\.crc\machines\crc\crc\Virtual Machines\9CD71052-C950-4AFB-9F51-3B6B5840F004.vmcx: The process cannot access the file because it is being used by another process.
exit status 1
```

**Comment by anjannath** [comment]
In the cleanup checks, we should first delete the VM and then proceed to delete the machine's directory, but why `Stop-Vm -Name crc -Force` hangs on windows 11 i am not sure

**Comment by adrianriobo** [comment]
related to #3726 as this is the situation where crc.exe process is not killed by `crc clenaup` (the daemon is started on a visible window)

**Comment by anjannath** [comment]
In Windows 11 the daemon preventing the cleanup will be tracked in #3726 

the `crc cleanup` command deleting the disk before the VM definition was fixed in #3728
